docs/en/index.md

```diff
@@ -159,33 +159,190 @@ Nested GridFields
 ```php
-class NestedObject extends DataObject
+class ChildNode extends DataObject
 {
-	private static $has_one = [
-		'Parent' => ParentObject::class
+    ...
+
+    private static $has_one = [
+        'ParentNode' => ParentNode::class,
+    ];
+
+    private static $has_many = [
+        'GrandChildNodes' => GrandChildNode::class,
+    ];
+
+    public function getNestedConfig(): GridFieldConfig
+    {
+        $config = new GridFieldConfig_RecordEditor();
+        $config->addComponent(GridFieldNestedForm::create()->setRelationName('GrandChildNodes'));
+        return $config;
+    }
+}
+
+```
+
+```php
+class GrandChildNode extends DataObject
+{
+    ...
+
+    private static $has_one = [
+        'ChildNode' => ChildNode::class,
+    ];
+}
+
+```
+
+```php
+class MyAdminSection extends ModelAdmin
+{
+    private static string $url_segment = 'my-admin-section';
+    private static string $menu_title = 'My Admin Section';
+
+    private static array $managed_models = [
+		ParentNode::class
 	];
 
-	public function getNestedConfig(): GridFieldConfig
-	{
-		$config = new GridFieldConfig_RecordViewer();
-		return $config;
-	}
+    protected function getGridFieldConfig(): GridFieldConfig
+    {
+        $config = parent::getGridFieldConfig();
+        $config->addComponent(GridFieldNestedForm::create()->setRelationName('ChildNodes'));
+
+        return $config;
+    }
 }
+
 ```
 
-You can also modify the default config (a `GridFieldConfig_RecordEditor`) via an extension to the nested model class, by implementing
-`updateNestedConfig`, which will get the config object as the first parameter.
+There is also the possibility to use Nested GridField with the data structure ArrayList. To do this, you can use the following approach.
+
+```php
+class MyDataSet extends DataObject
+{
+    ...
+
+    public function getMyArrayList() {
+        $list = ArrayList::create();
+        $data = MyDataObject::get();
+
+        foreach ($data as $value) {
+            $list->push($value);
+        }
+
+        return $list;
+    }
+}
+
+```
+
+```php
+class MyAdminSection extends ModelAdmin
+{
+    ...
+
+    private static array $managed_models = [
+		MyDataSet::class
+	];
+
+    protected function getGridFieldConfig(): GridFieldConfig
+    {
+        $config = parent::getGridFieldConfig();
+        $config->addComponent(GridFieldNestedForm::create()->setRelationName('getMyArrayList'));
+
+        return $config;
+    }
+}
+
+```
+
+
+#### Additional features and settings
+
+1. You can set the maximum number of nested levels using a `$default_max_nesting_level` configuration property. The default value is 10 levels.
+
+```yml
+Symbiote\GridFieldExtensions\GridFieldNestedForm:
+    default_max_nesting_level: 5
+```
+
+1. You can also modify the default config (a `GridFieldConfig_RecordEditor`) via an extension to the nested model class, by implementing
+`updateNestedConfig`, which will get the config object as the parameter.
+
```

Comment:
This isn't rendering correctly as a list. I think changing the indentation like this will work.
```suggestion
1. You can set the maximum number of nested levels using a `$default_max_nesting_level` configuration property. The default value is 10 levels.

    ```yml
    Symbiote\GridFieldExtensions\GridFieldNestedForm:
        default_max_nesting_level: 5
    ```

1. You can also modify the default config (a `GridFieldConfig_RecordEditor`) via an extension to the nested model class, by implementing
`updateNestedConfig`, which will get the config object as the parameter.

````

Reply:
Updated